From 1e245d822649ce603cd9cc1352306890747947b9 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 24 Sep 2020 06:50:53 -0700
Subject: [PATCH 1656/1851] v4l: xilinx: switch: Fix warnings for
 OVERFLOW_BEFORE_WIDEN

commit df32c6f8a78cdc811c79bd46417d80e1d5a51957 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the OVERFLOW_BEFORE_WIDEN type.

The actual warnings are:

For xvsw_s_stream():

Event  overflow_before_widen: Potentially overflowing expression "1 <<
xsw->nsources" with type "int" (32 bits, signed) is evaluated using
32-bit arithematic, and then used in a context that expects an
expression of type "unsigned long" (64 bits, unsigned).

Event remediation: To avoid overflow, cast "1" to type "unsigned long".

The checkpatch warns with this fix as below, but the coverity warning
sounds more reasonable. Hence ignoring the checkpatch warning for now.

WARNING: Unnecessary typecast of c90 int constant
+		   ((((unsigned long)1 << xsw->nsources) - 1) << 4) |

total: 0 errors, 1 warnings, 8 lines checked

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-switch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-switch.c b/drivers/media/platform/xilinx/xilinx-switch.c
index e249ac9b443c..e41c097e8b76 100644
--- a/drivers/media/platform/xilinx/xilinx-switch.c
+++ b/drivers/media/platform/xilinx/xilinx-switch.c
@@ -104,7 +104,7 @@ static int xsw_s_stream(struct v4l2_subdev *subdev, int enable)
 	xvip_write(&xsw->xvip, XSW_CORE_CH_CTRL, routing);
 
 	xvip_write(&xsw->xvip, XVIP_CTRL_CONTROL,
-		   (((1 << xsw->nsources) - 1) << 4) |
+		   ((((unsigned long)1 << xsw->nsources) - 1) << 4) |
 		   XVIP_CTRL_CONTROL_SW_ENABLE);
 
 	return 0;
-- 
2.31.1

